### DISCUSSION TITLE
Is there a Getting started resource on how to write new plugins?

### DISCUSSION THREAD

**Discussion by m6man** [extesy/hoverzoom#1030] [Q&A] [answered]
Hi, I'm a web developer and would like to add a plugin for the yiffer.xyz website. I've already looked at the code for the e621.net website, but it's not fully clear to me yet what's needed to make a new plugin for yiffer.xyz work. Is there a resource or guide that explains the most important parts of HZ+ plugins? Thank you.

**Comment by extesy** [comment]
The easiest approach would be to copy any existing plugin and adopt it to work for your website. You'd need to duplicate one of the files under the `plugins` directory and add it to `manifest.json`.

**Comment by GrosPoulet** [comment, marked as answer]
Hi @m6man,

In yiffer.xyz, each cover points to a comic (= a collection of several images).
So basically, for each cover you need to prepare a gallery (= urls of images contained in the comic) that HZ+ will display when user hovers the cover.
A good starting point would be the [bnf.js](https://github.com/extesy/hoverzoom/blob/master/plugins/bnf.js) plug-in i wrote.

e.g: for "Bifurcation" comic:
- cover url: https://yiffer.xyz/Bifurcation
- number of urls in gallery: `<p title="Number of pages">11</p>`
- urls: https://static.yiffer.xyz/comics/Bifurcation/001.jpg ... https://static.yiffer.xyz/comics/Bifurcation/011.jpg

**Reply by m6man** [reply]
Thanks for the response! I was able to make this PR: #1033 
My question was a bit more aimed at how the HZ+ API works, like how you must use the jQuery data storage and what the properties mean and all that. I was able to piece it together though and I think it's a decent first version. Taking a look at your bnf.js plugin definitely helped to get the gallery feature right.
